### ISSUE TITLE
npx install of grafana-toolkit broken

### ISSUE THREAD

**Issue by niclash** [grafana/grafana#46952]
I am trying to create a panel plugin, as mentioned on https://grafana.com/tutorials/build-a-panel-plugin/

(Related to #30887)

`d1` below is my development desktop and running Debian Testing. 
```
niclas@d1:~/dev/sensetif$ npm -v
8.5.5
niclas@d1:~/dev/sensetif$ node -v
v14.16.0
niclas@d1:~/dev/sensetif$ npx -v
7.10.0
```

So, running the `plugin:create` command;
```
niclas@d1:~/dev/sensetif$  npx @grafana/toolkit plugin:create sensetif-svg-plugin
```
And I get a message saying that dependencies can't be resolved (see below).

I have removed `$HOME/.npm` before running the command, and also checked if there is any relevant node modules installed globally, and I can't see that. Only a handful global ones, such as npm, yarn, grunt,...

Then I created a new VM on Hetzner with Ubuntu 20.04, to make sure it is totally reproducable by you guys and did the following;

```
root@grafana-test:~# apt update && apt upgrade && apt install npm
root@grafana-test:~# npx @grafana/toolkit plugin:create sensetif-svg-plugin
```

And that doesn't work at all, since Ubuntu 20.04 comes with node v 10.19.0 and that throws an exception for the `plugin:create` above;

```
  Trace: TypeError: lines.flat is not a function
      at Paginator.paginate (/root/.npm/_npx/30796/lib/node_modules/@grafana/toolkit/node_modules/inquirer/lib/utils/paginator.js:31:21)
      at ListPrompt.render (/root/.npm/_npx/30796/lib/node_modules/@grafana/toolkit/node_modules/inquirer/lib/prompts/list.js:121:31)
      at ListPrompt._run (/root/.npm/_npx/30796/lib/node_modules/@grafana/toolkit/node_modules/inquirer/lib/prompts/list.js:73:10)
      at Promise (/root/.npm/_npx/30796/lib/node_modules/@grafana/toolkit/node_modules/inquirer/lib/prompts/base.js:61:12)
      at new Promise (<anonymous>)
      at ListPrompt.run (/root/.npm/_npx/30796/lib/node_modules/@grafana/toolkit/node_modules/inquirer/lib/prompts/base.js:60:12)
:
:
```

So, everything around this is very fragile (two unrelated problems, on two very common Linux distros).

And here is the console output from my desktop which complains about dependencies;

```
niclas@d1:~/dev/sensetif$ npx @grafana/toolkit plugin:create sensetif-svg-plugin
Need to install the following packages:
  @grafana/toolkit
Ok to proceed? (y) 
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: slate@0.76.0
npm ERR! node_modules/slate
npm ERR!   peer slate@">=0.47.0" from @grafana/slate-react@0.22.10-grafana
npm ERR!   node_modules/@grafana/slate-react
npm ERR!     @grafana/slate-react@"0.22.10-grafana" from @grafana/ui@8.4.4
npm ERR!     node_modules/@grafana/ui
npm ERR!       @grafana/ui@"8.4.4" from @grafana/toolkit@8.4.4
npm ERR!       node_modules/@grafana/toolkit
npm ERR!         @grafana/toolkit@"*" from the root project
npm ERR!   peer slate@">=0.47.0" from slate-react-placeholder@0.2.9
npm ERR!   node_modules/@grafana/slate-react/node_modules/slate-react-placeholder
npm ERR!     slate-react-placeholder@"^0.2.8" from @grafana/slate-react@0.22.10-grafana
npm ERR!     node_modules/@grafana/slate-react
npm ERR!       @grafana/slate-react@"0.22.10-grafana" from @grafana/ui@8.4.4
npm ERR!       node_modules/@grafana/ui
npm ERR!         @grafana/ui@"8.4.4" from @grafana/toolkit@8.4.4
npm ERR!         node_modules/@grafana/toolkit
npm ERR!   2 more (slate-react, slate-plain-serializer)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer slate@">=0.32.0 <0.50.0" from slate-base64-serializer@0.2.115
npm ERR! node_modules/slate-base64-serializer
npm ERR!   slate-base64-serializer@"^0.2.111" from @grafana/slate-react@0.22.10-grafana
npm ERR!   node_modules/@grafana/slate-react
npm ERR!     @grafana/slate-react@"0.22.10-grafana" from @grafana/ui@8.4.4
npm ERR!     node_modules/@grafana/ui
npm ERR!       @grafana/ui@"8.4.4" from @grafana/toolkit@8.4.4
npm ERR!       node_modules/@grafana/toolkit
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/niclas/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/niclas/.npm/_logs/2022-03-25T10_16_02_386Z-debug.log
```

I will also attach the log file `/home/niclas/.npm/_logs/2022-03-25T10_16_02_386Z-debug.log`

**Comment by niclash** [comment]
I have also tried on Fedora 35, and there it works.

```
[root@grafana-test ~]# node -v && npm -v && npx -v
v16.14.0
8.3.1
8.3.1
```

Seeing matching `npm` and `npx` versions, MAY indicate the problem in my Desktop set up.

**Comment by niclash** [comment]
There was a `/usr/local/bin/npx` and no idea where it came from. 

After 
```
sudo apt remove nodejs
sudo rm /usr/local/bin/npx
sudo apt install nodejs
```
I got the `npx` to be version `8.5.5` as well. But my `nodejs` was back to `12.10`... Now what?

A bunch of `deprecated` messages, but otherwise it worked;

```
niclas@d1:~/dev/sensetif$ npx @grafana/toolkit plugin:create sensetif-svg-plugin
Need to install the following packages:
  @grafana/toolkit
Ok to proceed? (y) 
npm WARN ERESOLVE overriding peer dependency
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'react-dev-utils@12.0.0',
npm WARN EBADENGINE   required: { node: '>=14' },
npm WARN EBADENGINE   current: { node: 'v12.22.10', npm: '8.5.5' }
npm WARN EBADENGINE }
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated flatten@1.0.3: flatten is deprecated in favor of utility frameworks such as lodash.
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated @types/anymatch@3.0.0: This is a stub types definition. anymatch provides its own type definitions, so you do not need this installed.
npm WARN deprecated @braintree/sanitize-url@5.0.2: Potential XSS vulnerability patched in v6.0.0.
npm WARN deprecated html-webpack-plugin@3.2.0: 3.x is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated mkdirp@0.5.1: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
? Select plugin type (Use arrow keys)
❯ Panel Plugin 
  Datasource Plugin 
  Backend Datasource Plugin 
```

**Comment by niclash** [comment]
To close this off.

Installing latest version of Node on my system was not trivial, since all instructions are like this article; https://www.linuxcapable.com/how-to-install-node-js-14-lts-16-npm-on-debian-11-bullseye/

What this misses is that I have a `preferences` set differently, which placed `testing` at priority 900, whereas the `nodesource` repository would just get a default `500` priority, so `nodejs` would still be picked up from main debian repository.

Fixed that by adding a file `/etc/apt/preferences.d/nodesource.pref` with the following content;
```
Package: nodejs 
Pin: release n=bookworm
Pin-Priority: 920
```

Then remove+install `nodejs`.

Sorry for polluting the Grafana issue system.
